File: data/code_pairs/cpp/2033.cpp
// BoolArray.h
#include <iostream>

class BoolArray {
public:
    void subrBoolArrayIn(const bool* array_in, int n_in);
};

// BoolArray.cpp
#include "BoolArray.h"

void BoolArray::subrBoolArrayIn(const bool* array_in, int n_in) {
    std::cout << "SUBR_BOOLARRAY_IN" << std::endl;
    for (int i = 0; i < n_in; ++i) {
        std::cout << (i + 1) << " " << std::boolalpha << array_in[i] << std::endl;
    }
}

File: data/code_pairs/fortran/2033.f90
MODULE BOOLARRAY
  IMPLICIT NONE 
  PRIVATE
  PUBLIC :: SUBR_BOOLARRAY_IN
CONTAINS
  SUBROUTINE SUBR_BOOLARRAY_IN(ARRAY_IN, N_IN)
    INTEGER, INTENT(IN) :: N_IN
    LOGICAL, INTENT(IN) :: ARRAY_IN(:)
    INTEGER :: I
    PRINT *, 'SUBR_BOOLARRAY_IN'
    DO I = 1, N_IN
      PRINT *, I, ARRAY_IN(I)
    END DO
  END SUBROUTINE SUBR_BOOLARRAY_IN
END MODULE BOOLARRAY